libsolidity/lsp/LanguageServer.cpp

```diff
@@ -268,24 +268,17 @@ bool LanguageServer::run()
 	return m_state == State::ExitRequested;
 }
 
-bool LanguageServer::checkServerInitialized(MessageID _id)
+void LanguageServer::requireServerInitialized(MessageID _id)
 {
 	if (m_state != State::Initialized)
-	{
-		m_client.error(_id, ErrorCode::ServerNotInitialized, "Server is not properly initialized.");
-		return false;
-	}
-	else
-		return true;
+		throw HandlerError(_id, ErrorCode::ServerNotInitialized, "Server is not properly initialized.");
 }
 
 void LanguageServer::handleInitialize(MessageID _id, Json::Value const& _args)
 {
 	if (m_state != State::Started)
-	{
-		m_client.error(_id, ErrorCode::RequestFailed, "Initialize called at the wrong time.");
-		return;
-	}
+		throw HandlerError(_id, ErrorCode::RequestFailed, "Initialize called at the wrong time.");
```

Comment:
I think we can remove `_id` here and fill it out at the `catch`-site

Reply:
Also isn't it rather `RequestError`?

Reply:
And finally: Why not use `BOOST_THROW_EXCEPTION`?

Reply:
(or create a new `assert` macro)

Reply:
yes I had an issue in finding a good name for it. `RequestError` I thought about too. I'll pick it. wrt `_id`, yes, thx. I didn't use BOOST_THROW_EXCEPTION because it felt overkill, but i can use it.